src/utils/SpokeUtils.ts

```diff
@@ -205,6 +205,7 @@ export async function findFillBlock(
   const { provider } = spokePool;
   highBlockNumber ??= await provider.getBlockNumber();
   assert(highBlockNumber > lowBlockNumber, `Block numbers out of range (${lowBlockNumber} > ${highBlockNumber})`);
+  const { chainId: destinationChainId } = await spokePool.provider.getNetwork();
```

Comment:
You could also call spokePool.chainId() I believe, which might be less provider-opinionated

Reply:
I think it's preferable to use `provider.getNetwork()` because for a StaticJsonRpcProvider instance, the chainId is served directly out of the provider and never results in an external RPC call (nb. the `await` statement is misleading in this sense). Calling `SpokePool.chainId()` would instead result in an external call.